### DISCUSSION TITLE
Http client support

### DISCUSSION THREAD

**Discussion by helander** [boa-dev/boa#3076] [Q&A]
Does boa support the capability to act as a Javascript http client or is it possible to create some simple support. In the latter case, could you please provide some guidelines/hints on how to do that.

**Comment by jedel1043** [comment]
Boa is sandboxed by design: there are no network calls, fs calls or other side effects.

If you want to implement a simple HTTP server, you'd probably have to forward the API of a rust crate that offers that (maybe https://docs.rs/rouille/latest/rouille), and you could put your server state inside a `Class` implementor in order to access it from JS code (see https://github.com/boa-dev/boa/blob/main/boa_examples/src/bin/classes.rs)

**Reply by Qard** [reply]
An http server probably wouldn't make sense, but might make sense to have a plug-in providing the fetch API.

**Reply by Razican** [reply]
> An http server probably wouldn't make sense, but might make sense to have a plug-in providing the fetch API.

This is something that can be added to the boa_runtime crate :) feel free to ask for mentoring if you want to work on it!

**Reply by Qard** [reply]
I unfortunately don't have the time to contribute at the moment, but hoping to find the time for it eventually.

My focus currently is mainly on Node.js core, but I'm gradually experimenting with ideas for a specialized JS runtime of my own which I may end up converting to boa. If that works out the way I'm hoping I'd definitely get to contributing then. I'll for sure need to enable myself on a few things like snapshots and some serialization things. I'm currently working with V8 and making heavy use of snapshots and structured clone, though what I'm trying to do is proving to be a bit challenging. 🤔

**Comment by jedel1043** [comment]
Moved to Q&A in case people want to share their use cases on networking using Boa :)
